src/utils/logging.ts

```diff
@@ -0,0 +1,53 @@
+export interface Logging {
+  debug(msg: string): void;
+  warn(msg: string): void;
+  error(msg: string): void;
+  info(msg: string): void;
+}
+
+export enum LogLevel {
+  Debug,
+  Info,
+  Warn,
+  Error,
+}
+
+export class ConsoleLogger implements Logging {
+  public currentLogLevel;
+  private throwOnError;
+  private loggerName;
+
+  constructor(name: string, currentLogLevel: LogLevel, throwOnError: boolean) {
```

Comment:
You can simplify the code by changing the constructor signature to:

```ts
  constructor(readonly name: string, public currentLogLevel: LogLevel, private throwOnError = false) {
  }
```

And then you can also remove the property declaration in the class. Defining them with `readonly` / `private` / `public` in the constructor is equivalent to defining them as properties in the class and assigning them with the value from the constructor parameter. This is a [nice TypeScript shortcut](https://kendaleiv.com/typescript-constructor-assignment-public-and-private-keywords/).

Reply:
Fixed, first time doing typescript so happy to learn!